tests/unit/amo/reducers/test_addonsByAuthors.js

```diff
@@ -222,10 +324,144 @@ describe(__filename, () => {
+      // we end up looking through the entire reducer, so we have this filter
+      // to prevent an add-on from appearing in its own "by this author"
+      // list.
+      const addons = fakeAddons();
+      const authorNames = ['test', 'test2', 'test3'];
+      const state = reducer(undefined, loadAddonsByAuthors({
+        addons: Object.values(addons),
+        addonType: ADDON_TYPE_EXTENSION,
+        authorNames,
+      }));
+
+      expect(getAddonsForUsernames(
+        state, authorNames, ADDON_TYPE_EXTENSION, addons.firstAddon.slug
+      )).toEqual([
+        createInternalAddon(addons.secondAddon),
+        createInternalAddon(addons.thirdAddon),
+      ]);
+    });
+
+    it('returns nothing if no add-ons are found', () => {
+      const addons = fakeAddons();
+      const state = reducer(undefined, loadAddonsByAuthors({
+        addons: Object.values(addons),
+        authorNames: ['fakeUsername'],
+      }));
+
+      expect(getAddonsForUsernames(state, ['nobody'])).toBeNull();
+    });
+  });
+
+  describe('getLoadingForAuthorNames', () => {
+    it('returns loading for just authorNames', () => {
+      const state = reducer(undefined, fetchAddonsByAuthors({
+        authorNames: ['author1'],
+        errorHandlerId: 'error-handler-id',
+      }));
+
+      expect(getLoadingForAuthorNames(state, ['author1'])).toEqual(true);
+    });
+
+    it('returns loading for authorNames + addonType', () => {
+      const state = reducer(undefined, fetchAddonsByAuthors({
+        authorNames: ['author1'],
+        addonType: ADDON_TYPE_THEME,
+        errorHandlerId: 'error-handler-id',
+      }));
+
+      expect(getLoadingForAuthorNames(state, ['author1'], ADDON_TYPE_THEME))
+        .toEqual(true);
+    });
+
+    it('returns null when there is no match', () => {
+      const state = reducer(undefined, fetchAddonsByAuthors({
+        authorNames: ['author1'],
+        errorHandlerId: 'error-handler-id',
+      }));
+
+      expect(getLoadingForAuthorNames(state, ['author2'])).toEqual(null);
+    });
+
+    it('returns null when no authorNames provided', () => {
+      const state = reducer(undefined, fetchAddonsByAuthors({
+        authorNames: ['author1'],
+        errorHandlerId: 'error-handler-id',
+      }));
+
+      expect(getLoadingForAuthorNames(state, [])).toEqual(null);
+      expect(getLoadingForAuthorNames(state, null)).toEqual(null);
+    });
+  });
+
+  describe('joinAuthorNames', () => {
+    it('returns authorNames', () => {
+      expect(joinAuthorNames(['author1', 'author2']))
+        .toEqual('author1-author2');
+    });
+
+    it('returns authorNames + addonType', () => {
+      expect(joinAuthorNames(['author1', 'author2'], ADDON_TYPE_THEME))
+        .toEqual(`author1-author2-${ADDON_TYPE_THEME}`);
+    });
+
+    it('handles a single authorName', () => {
+      expect(joinAuthorNames(['author1'], ADDON_TYPE_THEME))
+        .toEqual(`author1-${ADDON_TYPE_THEME}`);
+    });
+  });
 });
```

Comment:
There don't seem to be any tests for the `forAuthorNamesAndAddonType` state variable.

Reply:
This also doesn't seem to have been addressed.